modules/terraform-aci-contract/main.tf

```diff
@@ -1,14 +1,15 @@
-
 locals {
   subj_filter_list = flatten([
     for subj in var.subjects : [
       for flt in coalesce(subj.filters, []) : {
-        id         = "${subj.name}-${flt.filter}"
-        subj       = subj.name
-        filter     = flt.filter
-        action     = flt.action
-        directives = join(",", concat(flt.log == true ? ["log"] : [], flt.no_stats == true ? ["no_stats"] : []))
-        priority   = flt.priority
+        id                    = "${subj.name}-${flt.filter}"
```

Comment:
You'll need to include the direction in the `id` otherwise you won't have unique keys if someone wants to use the same filter in each direction. This is an edge case but should be dealt with. This code should work:

`id = subj.apply_both_directions ? "${subj.name}-${flt.filter}" : "${subj.name}-${flt.filter}-${flt.direction}"`